### PR TITLE
fix: add directional hysteresis to snapResizeHeight (#2821)

### LINKED ISSUE #2821: Resize drag flickers at half-U boundaries: add directional hysteresis to snapResizeHeight
## Summary

snapResizeHeight rounds the drag distance to the nearest whole U (Math.round(growPx / pxPerU)). At the half-U line, pixel jitter flips the preview height back and forth, and each flip re-renders the rack frame, so the drag feels twitchy.

Design: https://github.com/RackulaLives/Rackula/blob/main/docs/superpowers/specs/2026-07-01-resize-bay-corrections-design.md

## Acceptance Criteria

- [ ] A resize step commits only after roughly 0.6U of travel past the current height in the direction of movement
- [ ] Holding the pointer at a step boundary produces no height oscillation
- [ ] Whole-U snapping and [minHeight, maxHeight] clamping are unchanged
- [ ] Keyboard resize (ArrowUp/ArrowDown) is unaffected

## Technical Notes

- src/lib/utils/rack-resize.ts: snapResizeHeight is pure. Extend its params with the current committed step so the threshold can be directional, for example: while (frac - (current - startHeight) >= 0.6) current++, mirrored for shrink.
- src/lib/components/RackCanvasView.svelte: handleResizeMove passes drag.previewHeight as the current step. handleResizeEnd recomputes from the release position and must apply the same rule so release cannot land one U off the preview.

## Test Requirements

- Unit tests at boundaries: 0.59U travel holds, 0.6U steps, reversing direction requires decisive travel back, min/max clamp interaction, degenerate pxPerU holds start height

### PR DESCRIPTION
**Pull request by ggfevans**
## Summary

`snapResizeHeight` rounded drag travel to the nearest whole U (`Math.round(growPx / pxPerU)`). At the half-U line, pixel jitter flipped the preview height back and forth, and each flip re-rendered the rack frame, so the drag felt twitchy.

This adds directional hysteresis. The threshold is measured from the current committed step (`currentHeight`): a step commits only after roughly 0.6U of travel past it in the direction of movement. That gives each step a +/-0.6U dead band, so the half-U flicker zone disappears. Whole-U stepping stays (rail invariant); continuous preview with snap-on-release was decided against in the epic #2820 spec and is not built here.

Both call sites in `RackCanvasView.svelte` pass the previewed height:

- `handleResizeMove` measures hysteresis from the step already on screen, so jitter at a U boundary does not flip the preview.
- `handleResizeEnd` applies the same rule from the same `previewHeight`, so a release cannot land one U off the height shown under the pointer.

## Acceptance criteria

- [x] A resize step commits only after roughly 0.6U of travel past the current height in the direction of movement
- [x] Holding the pointer at a step boundary produces no height oscillation
- [x] Whole-U snapping and [minHeight, maxHeight] clamping are unchanged
- [x] Keyboard resize (ArrowUp/ArrowDown) is unaffected (it never called `snapResizeHeight`)

## Invariant

Rail positions stay whole-U integers. The hysteresis is preview-step logic only: the committed height is seeded from `currentHeight` and only ever incremented/decremented by one, so it stays whole U. No fractional rail positions or committed heights are introduced.

## Tests

Added boundary unit tests for `snapResizeHeight` in `src/tests/rack-resize.test.ts` (TDD, written red then green):

- 0.59U travel holds; exactly 0.6U commits the next step
- No oscillation across the old half-U jitter band (40..59px at 100px/U all hold)
- A committed step stays steady while the pointer sits on its boundary
- Reversing direction requires 0.6U of decisive travel back
- Shrink threshold mirrors grow
- min/max clamp interaction from a committed step near the top and floor
- Degenerate pxPerU (0 and negative) holds the start height

The loops are bounded by the height range (not travel distance), the two loops are mutually exclusive, and output is integer-only under the integer input contract.

Gates run locally in the worktree: `npm run lint`, `npm run check`, `npm run test:run` (3149 tests pass), `npm run build`.

Closes #2821

Co-Authored-By: Claude Opus 4.8 (1M context) <noreply@anthropic.com>

🤖 Generated with [Claude Code](https://claude.com/claude-code)

___

## **CodeAnt-AI Description**
Make rack resizing stop flickering at U boundaries

### What Changed
- Drag resize now waits for a clear move past the current step before changing height, which prevents the preview from bouncing back and forth near half-U positions
- Releasing the drag now uses the same rule as the live preview, so the final height matches what the pointer was showing
- Resize still snaps to whole U values and still respects the minimum and maximum rack heights
- Added tests for boundary cases, including boundary jitter, reversing direction, shrink drags, and clamping at the limits

### Impact
`✅ Fewer resize flickers`
`✅ More stable drag previews`
`✅ Final resize height matches the preview`
<details><summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: src/tests/rack-resize.test.ts [hunk 1]
```diff
@@ -217,6 +217,149 @@ describe("rack-resize", () => {
...
+        snapResizeHeight({
+          ...base,
+          startHeight: 10,
+          growPx: 40,
+          pxPerU: 100,
+          currentHeight: 11,
+        }),
+      ).toBe(10);
+    });
+
```
[the remark this answers is not in the document]
Comment (ggfevans) [reply src/tests/rack-resize.test.ts:315]: Added in 761c10a: a mid-range multi-U jump test (3.7U from a non-boundary currentHeight lands on 14, exercising the while-loop step-skipping away from the clamp).
